### PR TITLE
Remove device SDK initializers that take connection string and device Id

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/helpers/TestDevice.cs [hunk 1]
```diff
@@ -194,12 +194,12 @@ public DeviceClient CreateDeviceClient(ITransportSettings[] transportSettings, C
...
                 {
                     deviceClient = DeviceClient.CreateFromConnectionString(ConnectionString, transportSettings, options);
-                    s_logger.Trace($"{nameof(CreateDeviceClient)}: Created {nameof(DeviceClient)} {Device.Id} from device connection string: ID={TestLogger.IdOf(deviceClient)}");
+                    s_logger.Trace($"{nameof(CreateDeviceClient)}: Created {nameof(DeviceClient)} {Device.Id} from device connection string: Id={TestLogger.IdOf(deviceClient)}");
                 }
                 else
                 {
-                    deviceClient = DeviceClient.CreateFromConnectionString(TestConfiguration.IoTHub.ConnectionString, Device.Id, transportSettings, options);
-                    s_logger.Trace($"{nameof(CreateDeviceClient)}: Created {nameof(DeviceClient)} {Device.Id} from IoTHub connection string: ID={TestLogger.IdOf(deviceClient)}");
+                    deviceClient = DeviceClient.CreateFromConnectionString($"{TestConfiguration.IoTHub.ConnectionString};DeviceId={Device.Id}", transportSettings, options);
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:201]: The one place we used it for testing is here, which we can simply do manually from the test helper.

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]
Are we maintaining a 1N of breaking changes that we're making? That'll be easier than scanning the entire sourcecode diff when writing the migration guide.

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**